fs/dafsserver/dafsserver.cpp

```diff
@@ -1178,10 +1183,18 @@ class CRemoteRequest : public CSimpleInterfaceOf<IInterface>
             activity->restoreCursor(cursorMb);
         }
 
+        __uint64 processStartTime = 0;
+        if (activeSpan->isValid())
+            processStartTime = nsTick();
+
         if (activity->queryIsReadActivity())
             processRead(requestTree, responseMb);
         else if (activity->queryIsWriteActivity())
             processWrite(requestTree, restMb, responseMb);
+
+        if (activeSpan->isValid())
+            processingTimeNS += (nsTick() - processStartTime);
```

Comment:
if there's an exception in process* then this won't happen.
Can you use a RAII pattern, so the time guaranteed to be accounted for?